documentation/specs/TextField.md

```diff
@@ -0,0 +1,307 @@
+- When `errorText` is supplied with `validationState="invalid"`, `helperText` is overriden.
+- Underlying input type can be set to `text`, `email`, `password`, `tel`, or `search`.
+- `isLabelVisuallyHidden` property can be used to visually hide the label.
+
+### Risks and Challenges
+
+- Accomodating various input types and variations while keeping a relatively simple and approachable API.
+- Ensuring the component remains accessible when label is visually hidden.
+
+### Prior Art
+
+- [Paste `<Input />`](https://paste.twilio.design/components/input)
+- [Polaris `<TextField />`](https://polaris.shopify.com/components/selection-and-input/text-field)
+- [Spectrum `<TextField />`](https://react-spectrum.adobe.com/react-spectrum/TextField.html)
+
+---
+
+## Design
+
+### API
+
+```ts
+import type { AriaTextFieldProps } from "react-aria";
+
+/**
+ * Note that this does not represent an exhaustive list of the props
+ * provided by AriaTextFieldProps, but they are notable properties that
+ * consumers may use on the TextField component
+ */
+export type ValidationState = "valid" | "invalid";
+
+type AriaTextFieldProps = {
+  /** Whether the input is disabled. */
+  isDisabled?: boolean;
+  /** Whether user input is required on the input before form submission. */
+  isRequired?: boolean;
+  /** Whether the input should display its "valid" or "invalid" visual styling. */
+  validationState?: ValidationState;
+  /** Whether the element should receive focus on render. */
+  autoFocus?: boolean;
+  /** Temporary text that occupies the text input when it is empty. */
+  placeholder?: string;
+  /** The current value (controlled). */
+  value?: string;
+  /** The default value (uncontrolled). */
+  defaultValue?: string;
+  /** The content to display as the label. */
+  label: ReactNode;
+  /** Handler that is called when the value changes. */
+  onChange?: (value: C) => void;
+};
+
+export type InputType = "text" | "email" | "password" | "tel" | "search";
+export type TextFieldVariant = "protected" | "standard";
+export type TextFieldSize = "sm" | "md" | "lg";
+
+export type TextFieldProps = AriaTextFieldProps & {
+  /**
+   * Underlying HTML input type
+   * @default 'text'
+   */
+  type?: InputType;
+  /**
+   * When set to `protected`, the underlying input is set to `password` and a right aligned
+   * clickable and focusable visibility icon is added. The net effect is to give a PasswordInput.
+   * @default 'standard'
+   */
+  variant?: TextFieldVariant;
+  /**
+   * TextField size affects the overall size of the input, but it also influences the size of
+   * leftAlignedIcon and rightAlignedIcon. For instance, with `sm` size, the associated leftAlignedIcon
+   * or rightAlignedIcon size maps to the Easy UI icon size of 'sm'.
+   * @default 'md'
+   */
+  size?: TextFieldSize;
+  /**
+   * Visually hides the label, but keeps it accessible.
+   * @default false
+   */
+  isLabelVisuallyHidden?: boolean;
```

Comment:
per design, its a valid use case to have inputs without labels. from my understanding to keep it accessible, we can either visually hide the label (seems like we have a mixin in easy-ui already to do this) or add an `aria-label` attribute to the input (consumer provides the value).